### PR TITLE
Complete `lib/query.js` tests

### PR DESCRIPTION
**Pull request by rmeritz**
- Mostly just extend the existing `select` tests which actually use the query's functionality
- Add unit tests for small corner error cases that are hard to capture elsewhere
- Add one ignore for a else that will never be hit because of `callbackToPromise`

### CODE REVIEW

File: test/query.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+var Query = require('../lib/query');
+
+describe('Query', function() {
```
Comment (jugglinmike) [inline test/query.test.js:5]: I get the sense that `Query` is not a public API, so these tests may be a hindrance during refactoring. What do you think about verifying this behavior through `Table#select`?

File: test/select.test.js [hunk 2]
```diff
@@ -49,6 +49,255 @@ describe('record selection', function() {
...
+                        expect(records.length).toBe(1);
+                        records.forEach(function(record) {
+                            expect(record.id).toBe('recordB');
+                            expect(record.get('Name')).toBe('Clinton');
+                        });
+                    }
+                    iterationCounter++;
+                    fetchNextPage();
+                }).then(function() {
+                    expect(iterationCounter).toBe(2);
```
Comment (jugglinmike) [inline test/select.test.js:234]: Sorry for the noisy CI report; we probably don't need to lint across versions of Node.js. [This patch](https://github.com/Airtable/airtable.js/pull/190) ought to help.

File: test/query.test.js (hunk 1 above)
Comment (rmeritz) [reply test/query.test.js:5]: I am testing most things in `select` because I see at as likely being the public API. 

The handful of error states I'm addressing here are actually impossible to reach via select because select does its own validation and parameter filtering and error catching before passing the query directly to select. So its not possible to have 100% code coverage with out these unit tests. See `lib/table:49` Should I remove them and add ignore? Or add a note?

I could move the `validates cellFormat params` but that feels like an okay low level unit test to me.

File: test/select.test.js (hunk 2 above)
Comment (rmeritz) [reply test/select.test.js:234]: Thanks!

File: test/query.test.js (hunk 1 above)
Comment (jugglinmike) [reply test/query.test.js:5]: If `Query` is truly private, then any code paths that can't be reached in production are dead.

We could certainly merge the tests even if that's true. They're correct, after all. My only concern is that, in the absence of documentation, the presence of tests may lead others to believe that the code they exercise is important.

@kasra We believe that `Query` is private. Could you confirm?

Comment (kasrak) [reply test/query.test.js:5]: Yep `Query` is intended to be private, so would prefer to not add tests specifically for it. I think it would be reasonable to remove the extra error checking in the `Query` constructor if they're not reachable, but let's also update the comment for the `Query` constructor to document that it expects the caller to first validate `params` by calling `Query.validateParams`.

Comment (rmeritz) [reply test/query.test.js]: I completely removed and added comments about all of the unreachable error conditions which allowed me to completely remove the `query.test.js` file.

File: lib/query.js [hunk 3]
```diff
@@ -93,9 +84,7 @@ function eachPage(pageCallback, done) {
                     next = inner;
                 } else {
                     next = function() {
-                        if (done) {
-                            done(null);
-                        }
+                        done(null);
```
Comment (kasrak) [inline lib/query.js:87]: The check on line 68 and corresponding error message on line 69 says it's valid for `done` to be `undefined`, which would make this line crash. Is that not possible because of `callbackToPromise`? If so, we should update the check on line 68-69

Comment (rmeritz) [reply lib/query.js:87]: The end use of `eachPage` can allow `done` to be `undefined` and in that case it will be defined by `this.eachPage = callbackToPromise(eachPage, this, 1);` on query line 30. So it is an error message that make sense from an end user perspective.

Comment (kasrak) [reply lib/query.js:87]: Ah gotcha, thanks for the explanation
